WordPressKit/AccountServiceRemoteREST+SocialService.swift

```diff
@@ -12,19 +13,35 @@ extension AccountServiceRemoteREST {
     /// - Parameters:
     ///     - service The name of the social service.
     ///     - token The OpenID Connect (JWT) ID token identifying the user on the social service.
+    ///     - appleEmail The email address associated with an Apple ID. Optional.
+    ///     - appleFullName The full name associated with an Apple ID. Optional.
     ///     - oAuthClientID The WPCOM REST API client ID.
     ///     - oAuthClientSecret The WPCOM REST API client secret.
     ///     - success The block that will be executed on success.
     ///     - failure The block that will be executed on failure.
-    public func connectToSocialService(_ service: SocialServiceName, serviceIDToken token: String, oAuthClientID: String, oAuthClientSecret: String, success:@escaping (() -> Void), failure:@escaping ((NSError) -> Void)) {
+    public func connectToSocialService(_ service: SocialServiceName,
+                                       serviceIDToken token: String,
+                                       appleEmail: String? = nil,
```

Comment:
I feel like these apple parameters don't really belong in here 🤔 The method can be used for any service, so having apple specific parameters there if you're using Google doesn't feel quite right to me (even though you can ignore them because of the defaults).

How about if we just accept a parameters dictionary, which we can append to our own `params`? We could add a separate method to generate it, like `appleSignInParameters(email:fullName:)`, so the client doesn't need to know the correct keys for those items.

Reply:
Or we could just pass in `userEmail` and `userFullName`, and the services can do with them what they want (or ignore them)